worker/mutation.go

```diff
@@ -424,6 +424,32 @@ func checkSchema(s *pb.SchemaUpdate) error {
 			x.ParseAttr(s.Predicate))
 	}
 
+	if s.Unique {
+		if s.ValueType == pb.Posting_STRING {
```

Comment:
unique directive does not mean that the predicate cannot have term index. It just means that we should also have the hash or exact index. We should prefer the hash index.